internal/provider/namespace_export_sink_resource.go

```diff
@@ -0,0 +1,552 @@
+		RoleName     types.String `tfsdk:"role_name"`
+		BucketName   types.String `tfsdk:"bucket_name"`
+		Region       types.String `tfsdk:"region"`
+		KmsArn       types.String `tfsdk:"kms_arn"`
+		AwsAccountId types.String `tfsdk:"aws_account_id"`
+	}
+
+	gcsSpecModel struct {
+		SaId         types.String `tfsdk:"sa_id"`
+		BucketName   types.String `tfsdk:"bucket_name"`
+		GcpProjectId types.String `tfsdk:"gcp_project_id"`
+		Region       types.String `tfsdk:"region"`
+	}
+)
+
+func NewNamespaceExportSinkResource() resource.Resource {
+	return &namespaceExportSinkResource{}
+}
+
+var (
+	_ resource.Resource                = (*namespaceExportSinkResource)(nil)
+	_ resource.ResourceWithConfigure   = (*namespaceExportSinkResource)(nil)
+	_ resource.ResourceWithImportState = (*namespaceExportSinkResource)(nil)
+
+	namespaceExportSinkSpecModelAttrTypes = map[string]attr.Type{
+		"name":    types.StringType,
+		"enabled": types.BoolType,
+		"s3": types.ObjectType{
+			AttrTypes: s3SpecModelAttrTypes,
+		},
+		"gcs": types.ObjectType{
+			AttrTypes: gcsSpecModelAttrTypes,
+		},
+	}
+
+	s3SpecModelAttrTypes = map[string]attr.Type{
+		"role_name":      types.StringType,
+		"bucket_name":    types.StringType,
+		"region":         types.StringType,
+		"kms_arn":        types.StringType,
+		"aws_account_id": types.StringType,
+	}
+
+	gcsSpecModelAttrTypes = map[string]attr.Type{
+		"sa_id":          types.StringType,
+		"bucket_name":    types.StringType,
+		"gcp_project_id": types.StringType,
+		"region":         types.StringType,
+	}
+)
+
+func (r *namespaceExportSinkResource) Configure(ctx context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	if req.ProviderData == nil {
+		return
+	}
+
+	client, ok := req.ProviderData.(*client.Client)
+	if !ok {
+		resp.Diagnostics.AddError(
+			"Unexpected Data Source Configure Type",
+			fmt.Sprintf("Expected *client.Client, got: %T. Please report this issue to the provider developers.", req.ProviderData),
+		)
+		return
+	}
+
+	r.client = client
+}
+
+func (r *namespaceExportSinkResource) Metadata(ctx context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_namespace_export_sink"
+}
+
+func (r *namespaceExportSinkResource) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Description: "Provisions a namespace export sink.",
+		Attributes: map[string]schema.Attribute{
+			"namespace": schema.StringAttribute{
+				Description: "The namespace under which the sink is configured. It's needed to be in the format of <namespace>.<account_id>",
+				Required:    true,
+			},
+			"id": schema.StringAttribute{
+				Description: "The unique identifier of the namespace export sink.",
+				Computed:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+			"spec": schema.SingleNestedAttribute{
```

Comment:
We have not been following the pattern of nesting the settings under a `spec` object in the other resources. I think for consistency we should go ahead and remove this wrapper.